server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -573,4 +465,192 @@ public TombstoneDocSupplier getTombstoneDocSupplier() {
+            this.eventListener = eventListener;
+            return this;
+        }
+
+        public Builder setQueryCache(QueryCache queryCache) {
+            this.queryCache = queryCache;
+            return this;
+        }
+
+        public Builder setQueryCachingPolicy(QueryCachingPolicy queryCachingPolicy) {
+            this.queryCachingPolicy = queryCachingPolicy;
+            return this;
+        }
+
+        public Builder setTranslogConfig(TranslogConfig translogConfig) {
+            this.translogConfig = translogConfig;
+            return this;
+        }
+
+        public Builder setFlushMergesAfter(TimeValue flushMergesAfter) {
+            this.flushMergesAfter = flushMergesAfter;
+            return this;
+        }
+
+        public Builder setExternalRefreshListener(List<ReferenceManager.RefreshListener> externalRefreshListener) {
+            this.externalRefreshListener = externalRefreshListener;
+            return this;
+        }
+
+        public Builder setInternalRefreshListener(List<ReferenceManager.RefreshListener> internalRefreshListener) {
+            this.internalRefreshListener = internalRefreshListener;
+            return this;
+        }
+
+        public Builder setIndexSort(Sort indexSort) {
+            this.indexSort = indexSort;
+            return this;
+        }
+
+        public Builder setCircuitBreakerService(CircuitBreakerService circuitBreakerService) {
+            this.circuitBreakerService = circuitBreakerService;
+            return this;
+        }
+
+        public Builder setGlobalCheckpointSupplier(LongSupplier globalCheckpointSupplier) {
+            this.globalCheckpointSupplier = globalCheckpointSupplier;
+            return this;
+        }
+
+        public Builder setRetentionLeasesSupplier(Supplier<RetentionLeases> retentionLeasesSupplier) {
+            this.retentionLeasesSupplier = retentionLeasesSupplier;
+            return this;
+        }
+
+        public Builder setPrimaryTermSupplier(LongSupplier primaryTermSupplier) {
+            this.primaryTermSupplier = primaryTermSupplier;
+            return this;
+        }
+
+        public Builder setTombstoneDocSupplier(TombstoneDocSupplier tombstoneDocSupplier) {
+            this.tombstoneDocSupplier = tombstoneDocSupplier;
+            return this;
+        }
+
+        public Builder setTranslogDeletionPolicyFactory(TranslogDeletionPolicyFactory translogDeletionPolicyFactory) {
+            this.translogDeletionPolicyFactory = translogDeletionPolicyFactory;
+            return this;
+        }
+
+        public Builder setIsReadOnlyReplica(boolean isReadOnlyReplica) {
+            this.isReadOnlyReplica = isReadOnlyReplica;
+            return this;
+        }
+
+        public Builder setTranslogFactory(TranslogFactory translogFactory) {
+            this.translogFactory = translogFactory;
+            return this;
+        }
+
+        public EngineConfig createEngineConfig() {
+            return new EngineConfig(
```

Comment:
Another nitpick...assuming this constructor can be made private, then you can pass the `this` instance of EngineConfig.Builder and let the constructor read the fields. It's a bit more concise and makes adding a new field require fewer changes.

Reply:
I like this. Will change in next commit.

Reply:
+1 for the review comment from @andrross. Without a private constructor here, the builder for `EngineConfig` can be bypassed.